book_source/06_reference/04_docker/03_architecture.Rmd

```diff
@@ -17,5 +19,186 @@ The PEcAn framework containers consist of multiple unique ways to interact with
+The services comprising the PEcAn application are described below.
+
+- `networks` -- This is a list of networks used by the application.
+We just provide the network name (`pecan`) and resort to Docker's default network configuration.
+Note that the services we want connected to this network include a `networks: ... - pecan` tag.
+For more details on Docker networks, see the [official documentation](https://docs.docker.com/network/).
+
+- `volumes` -- Similarly to `networks`, this just contains a list of volume names we want.
+Briefly, in Docker, volumes are directories containing files that are meant to be shared across containers.
+Each volume corresponds to a directory, which can be mounted at a specific location by different containers.
+For example, syntax like `volumes: ... - pecan:/data` in a service definition means to mount the `pecan` "volume" (including its contents) in the `/data` directory of that container.
+For more details on Docker volumes, see the [official documentation](https://docs.docker.com/storage/volumes/).
+Here, we define three volumes:
+
+	- `postgres` -- This contains the data files underlying the PEcAn PostgreSQL database (BETY).
+	Notice that it is mounted by the `postgres` container to `/var/lib/postgresql/data`.
+	This is the data that we pre-populate when we run the Docker commands to [initialize the PEcAn database](#pecan-docker-quickstart-init).
+	Note that these are the values stored _directly in the PostgreSQL database_.
+	The default files to which the database points (i.e. `dbfiles`) are stored in the `pecan` volume, described below.
+	
+	- `rabbitmq` -- This volume contains persistent data for RabbitMQ.
+	It is only used by the `rabbitmq` service.
+	
+	- `pecan` -- This volume contains PEcAn's `dbfiles`, which include downloaded and converted model inputs, processed configuration files, and outputs.
+	It is used by almost all of the services in the PEcAn stack, and is typically mounted to `/data`.
+ 
+#### `traefik` {#pecan-dc-traefik}
+
+[Traefik](https://traefik.io/) manages communication among the different PEcAn services and between PEcAn and the web.
+Among other things, `traefik` facilitates the setup of web access to each PEcAn service via common and easy-to-remember URLs.
+For instance, the following lines in the `web` service configure access to the PEcAn web interface via the URL http://localhost:8000/pecan/ :
+
+```{r, echo = FALSE, comment = ''}
+yaml::write_yaml(dc_yaml$services$web["labels"], stdout())
+```
+
+(Further details in the works...)
+
+The traefik service configuration looks like this:
+
+```{r, echo = FALSE, comment = ''}
+yaml::write_yaml(dc_yaml$services["traefik"], stdout())
+```
+
+#### `minio` {#pecan-dc-minio}
+
+[Minio](https://github.com/minio/minio) is a browser-based file manager that facilitates exploration of PEcAn files without the need for CLI tools.
```

Comment:
minio is service that will provide access to the a folder on disk. This data is made available using different protocol, such as S3 buckets, web based, etc. We mainly use this service to allow for easy access to the data using a web browser.

Reply:
Addressed in b85069e35.